mass_mailing_list_dynamic/README.rst

```diff
@@ -0,0 +1,79 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+   :target: https://www.gnu.org/licenses/agpl
+   :alt: License: AGPL-3
+
+==========================
+Dynamic Mass Mailing Lists
+==========================
+
+Without this addon you have to choose between providing a dynamic domain and
+letting your mass mailings reach all partners that match it, or being able to
+unsubscribe to certain mailing lists while still being subscribed to others.
+
+This addon allows you to create dynamic mailing lists, so you can now benefit
+from both things.
+
+Usage
+=====
+
+To create a dynamic mailing list, you need to:
+
+#. Go to *Mass Mailing > Mailings > Mailing Lists > Create*.
+#. Choose a *Sync method*.
+#. Choose a *Sync domain*.
+
+When you hit the *Sync now* button or send a mass mailing to this list, its
+contacts will be automatically updated.
+
+Pay attention to the messages shown to you that tell you about some non-obvious
+behaviour you could experience if you edit manually contacts from a dynamic
+list.
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/205/10.0
+
+Known issues / Roadmap
+======================
+
+* Tests affected by https://github.com/odoo/odoo/issues/20603. Do not run them
+  in stateful databases.
+* Forbid users about adding or removing contacts to fully synchronized lists.
```

Comment:
Please implement this. You can do it intercepting `create` method of a list contact and checking sync_method == full sync plus the flag that is still missing on the owner list. This way, you don't need to make UI tricks and you cover also the programmatic case. This only requires a context key to be checked for allowing to add contacts when running sync method.